.github/workflows/main.yml

```diff
@@ -33,31 +59,62 @@ jobs:
 
   build-n-test-n-coverage:
     name: Build, test and code coverage
+
+    needs: check_duplicate_runs
+    if: ${{ needs.check_duplicate_runs.outputs.should_skip != 'true' }}
+
     runs-on: ${{ matrix.os }}
+
     strategy:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
         python-version: [3.6, 3.7, 3.8, 3.9]
+
     env:
       OS: ${{ matrix.os }}
-      PYTHON: "3.9"
 
     steps:
       - name: Checkout
         uses: actions/checkout@v2
+
       - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v1
+        uses: actions/setup-python@v2
```

Comment:
This is very old action tag. Also v2 is already used in pre-commit step, so I've updated it.